src/watertap_contrib/seto/analysis/multiperiod/vagmd_batch/VAGMD_batch_design_model.py

```diff
@@ -0,0 +1,393 @@
+):
+
+    # Identify if the final brine salinity is larger than 175.3 for module "AS7C1.5L"
+    final_brine_salinity = feed_salinity / (1 - recovery_ratio)  # g/L
+    if module_type == "AS7C1.5L" and final_brine_salinity > 175.3:
+        cooling_system_type = "closed"
+        feed_flow_rate = 1100
+        evap_inlet_temp = 80
+        cond_inlet_temp = 25
+        high_brine_salinity = True
+    else:
+        high_brine_salinity = False
+
+    # Identify the module area
+    if module_type == "AS7C1.5L":
+        module_area = 7.2
+    else:  # module_type = "AS26C7.2L"
+        module_area = 25.92
+
+    # Identify the time step of the simulation (second)
+    if module_type == "AS7C1.5L":
+        dt = 20352.55 / feed_flow_rate
+    else:  # module_type == "AS26C7.2L"
+        dt = 73269.19 / feed_flow_rate
+
+    initial_status = _get_membrane_performance(
+        evap_inlet_temp,
+        feed_flow_rate,
+        cond_inlet_temp,
+        feed_salinity,
+        module_type,
+        high_brine_salinity,
+    )
+
+    PFlux = [initial_status[0]]  # Permeate flux, kg/h/m2
+    TCO = [initial_status[1]]  # Condenser outlet temperature, oC
+    TEO = [initial_status[2]]  # Evaporator outlet temperature, oC
+    TCI = [cond_inlet_temp]  # Condenser inlet temperature, oC
+    Ttank = [feed_temp]  # Feed temperature in thetank, oC
+    V = [initial_batch_volume]  # Volume remaining in the tank, L
+    Vd = [0]  # Volume of distillate, L
+    PFR = [PFlux[0] * module_area]  # Permeate flow rate, L/h
+    AccVd = [0]  # Accumulated volume of distillate, L
+    RR = [0]  # Accumlated recovery ratio, dimensionless
+    S = [feed_salinity]  # Feed salinity in the tank, g/L
+
+    while S[-1] < final_brine_salinity:
+        Vd.append(PFR[-1] * dt / 3600)
+        V.append(V[-1] - Vd[-1])
+        S.append(V[-2] / V[-1] * S[-1])
+        Ttank.append(
+            (feed_flow_rate * dt / 3600 * TEO[-1] + V[-2] * Ttank[-1])
+            / (V[-2] + feed_flow_rate * dt / 3600)
+        )
+
+        if cooling_system_type == "closed":
+            TCI.append(TCI[-1])
+
+        current_status = _get_membrane_performance(
+            evap_inlet_temp,
+            feed_flow_rate,
+            TCI[-1],
+            S[-1],
+            module_type,
+            high_brine_salinity,
+        )
+
+        PFlux.append(current_status[0])
+        TCO.append(current_status[1])
+        TEO.append(current_status[2])
+        PFR.append(PFlux[-1] * module_area)
+        AccVd.append(Vd[-1] + AccVd[-1])
+        RR.append(
+            (
+                1
+                - (
+                    (feed_flow_rate - PFR[-1])
+                    / feed_flow_rate
+                    * (S[-1] / S[0])
+                    * (1 - RR[-1])
+                )
+            )
+        )
+
+    return len(PFlux)
+
+
+def _get_membrane_performance(TEI, FFR, TCI, SgL, module_type, high_brine_salinity):
+    # Model parameters
+    PFluxAS26 = [
```

Comment:
I would rename these lists of model parameters (e.g., `PfluxAS26`) to more readable names, or at least describe them in a doc string under `_get_membrane_performance`

Reply:
In fact, you should add a doc string that explains the args of this function too

Reply:
Added and will be there in the new PR.